<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="A.DeMartin" />

<meta name="date" content="2025-03-17" />

<title>CUP Umaps</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CUP</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">CUP Umaps</h1>
<h4 class="author">A.DeMartin</h4>
<h4 class="date">2025-03-17</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-05-02
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>CUP/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250318code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250318)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250318code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250318)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/R/CUP/data/CD45neg_allmerged_seurat.rds
</td>
<td style="text-align:left;">
data/CD45neg_allmerged_seurat.rds
</td>
</tr>
<tr>
<td style="text-align:left;">
/R/CUP/analysis/markerGenes
</td>
<td style="text-align:left;">
analysis/markerGenes
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomHWC0406CUPtreee1bdb0b292ad4fdded81e5b38224bf7970f1063etargetblanke1bdb0ba">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/HWC0406/CUP/tree/e1bdb0b292ad4fdded81e5b38224bf7970f1063e" target="_blank">e1bdb0b</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomHWC0406CUPtreee1bdb0b292ad4fdded81e5b38224bf7970f1063etargetblanke1bdb0ba"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/HWC0406/CUP/tree/e1bdb0b292ad4fdded81e5b38224bf7970f1063e" target="_blank">e1bdb0b</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    data/CD45pos/
    Ignored:    data/seurat_objects/

Untracked files:
    Untracked:  analysis/CUP_knitting_easy_read.Rmd
    Untracked:  analysis/markerGenes_CD45pos

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level2">
<h2>load packages</h2>
</div>
<div id="load-file" class="section level2">
<h2>load file</h2>
<pre class="r"><code>##load merged file 
fileNam &lt;- &quot;/R/CUP/data/CD45neg_allmerged_seurat.rds&quot;
seuratM &lt;- readRDS(fileNam)
table(seuratM$dataset)</code></pre>
<pre><code>
          1_20200909_Human__LN_FSC1_v3            1_20210113_Human_LN_FSC1_v3 
                                  2588                                   1558 
      13_20200723_Hu_LN_Lin_neg_FSC_v3           2_20200909_Human__LN_FSC2_v3 
                                  3133                                   2606 
           2_20210113_Human_LN_FSC2_v3 373291_01-1_20241211_Hu_CUP_01_CD45neg 
                                  1019                                   2264 
373291_03-3_20241211_Hu_CUP_04_CD45neg 373291_05-5_20241211_Hu_CUP_05_CD45neg 
                                  5540                                   4708 
373291_07-7_20241211_Hu_CUP_03_CD45neg  380131_01-1_20250224_Hu_CUP_01_stroma 
                                  1718                                   3758 
o27533_1_01-1_20220201_Hu_LN_stroma_v3 o27533_1_05-5_20220202_Hu_LN_stroma_v3 
                                  9001                                   7613 </code></pre>
<pre class="r"><code>table(seuratM$RNA_snn_res.0.25)</code></pre>
<pre><code>
   0    1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16   17   18   19 
8491 4373 3754 3148 3140 3121 2506 2431 2029 1946 1890 1855 1505 1113 1036 1036  896  592  543  101 </code></pre>
<pre class="r"><code>table(seuratM$orig.ident)</code></pre>
<pre><code>
      
45506 </code></pre>
<pre class="r"><code>### add any type of metadata
### patient
CUP_01 &lt;- c(&quot;373291_01-1_20241211_Hu_CUP_01_CD45neg&quot;, &quot;380131_01-1_20250224_Hu_CUP_01_stroma&quot;)
CUP_03 &lt;- c(&quot;373291_07-7_20241211_Hu_CUP_03_CD45neg&quot;)
CUP_04 &lt;- c(&quot;373291_03-3_20241211_Hu_CUP_04_CD45neg&quot;)
CUP_05 &lt;- c(&quot;373291_05-5_20241211_Hu_CUP_05_CD45neg&quot;)
hLN_01 &lt;- c(&quot;13_20200723_Hu_LN_Lin_neg_FSC_v3&quot;)
hLN_02 &lt;- c(&quot;1_20200909_Human__LN_FSC1_v3&quot;,&quot;2_20200909_Human__LN_FSC2_v3&quot;)
hLN_03 &lt;- c(&quot;1_20210113_Human_LN_FSC1_v3&quot;,&quot;2_20210113_Human_LN_FSC2_v3&quot;)
hLN_04 &lt;- c(&quot;o27533_1_01-1_20220201_Hu_LN_stroma_v3&quot;)
hLN_05 &lt;- c(&quot;o27533_1_05-5_20220202_Hu_LN_stroma_v3&quot;)

seuratM$patient &lt;- &quot;pat_nr&quot;
seuratM$patient[which(seuratM$dataset %in% CUP_01)] &lt;- &quot;CUP_01&quot;
seuratM$patient[which(seuratM$dataset %in% CUP_03)] &lt;- &quot;CUP_03&quot;
seuratM$patient[which(seuratM$dataset %in% CUP_04)] &lt;- &quot;CUP_04&quot;
seuratM$patient[which(seuratM$dataset %in% CUP_05)] &lt;- &quot;CUP_05&quot;
seuratM$patient[which(seuratM$dataset %in% hLN_01)] &lt;- &quot;hLN_01&quot;
seuratM$patient[which(seuratM$dataset %in% hLN_02)] &lt;- &quot;hLN-02&quot;
seuratM$patient[which(seuratM$dataset %in% hLN_03)] &lt;- &quot;hLN_03&quot;
seuratM$patient[which(seuratM$dataset %in% hLN_04)] &lt;- &quot;hLN_04&quot;
seuratM$patient[which(seuratM$dataset %in% hLN_05)] &lt;- &quot;hLN_05&quot;
table(seuratM$patient)</code></pre>
<pre><code>
CUP_01 CUP_03 CUP_04 CUP_05 hLN-02 hLN_01 hLN_03 hLN_04 hLN_05 
  6022   1718   5540   4708   5194   3133   2577   9001   7613 </code></pre>
<pre class="r"><code>###order patients
ordpatients &lt;- c(&quot;hLN_01&quot;,&quot;hLN-02&quot;, &quot;hLN_03&quot;,&quot;hLN_04&quot;, &quot;hLN_05&quot;,&quot;CUP_01&quot;, &quot;CUP_03&quot;, &quot;CUP_04&quot;,&quot;CUP_05&quot;)
seuratM$patient &lt;- factor(seuratM$patient, levels=c(&quot;hLN_01&quot;,&quot;hLN-02&quot;, &quot;hLN_03&quot;,&quot;hLN_04&quot;, &quot;hLN_05&quot;,&quot;CUP_01&quot;, &quot;CUP_03&quot;, &quot;CUP_04&quot;,&quot;CUP_05&quot;))
Idents(seuratM) &lt;- seuratM$patient
table(seuratM$patient)</code></pre>
<pre><code>
hLN_01 hLN-02 hLN_03 hLN_04 hLN_05 CUP_01 CUP_03 CUP_04 CUP_05 
  3133   5194   2577   9001   7613   6022   1718   5540   4708 </code></pre>
<pre class="r"><code>#### diseaseCond
seuratM$diseaseCond &lt;- &quot;diseaseCond&quot;
seuratM$diseaseCond[grepl(&quot;CUP&quot;, seuratM$dataset)] &lt;- &quot;CUP&quot;
seuratM$diseaseCond[grepl(&quot;LN&quot;, seuratM$dataset)] &lt;- &quot;healthy&quot;
table(seuratM$diseaseCond)</code></pre>
<pre><code>
    CUP healthy 
  17988   27518 </code></pre>
<pre class="r"><code>###order
orddiseaseCond &lt;- c(&quot;healthy&quot;, &quot;CUP&quot;)
seuratM$diseaseCond &lt;- factor(seuratM$diseaseCond, levels=c(&quot;healthy&quot;, &quot;CUP&quot;))
Idents(seuratM) &lt;- seuratM$diseaseCond
table(seuratM$diseaseCond)</code></pre>
<pre><code>
healthy     CUP 
  27518   17988 </code></pre>
<pre class="r"><code>## Remove outlier and filter out CD45+ clusters 

seuratMfil &lt;- subset(seuratM, patient == c(&quot;hLN_01&quot;), invert = TRUE) 

seuratMfil &lt;- subset(seuratMfil, RNA_snn_res.0.25 %in% c(&quot;7&quot;,&quot;9&quot;,&quot;11&quot;,&quot;19&quot;),invert = TRUE)

table(seuratMfil$patient)</code></pre>
<pre><code>
hLN_01 hLN-02 hLN_03 hLN_04 hLN_05 CUP_01 CUP_03 CUP_04 CUP_05 
     0   5127   2418   8582   7490   5790   1296   5219    118 </code></pre>
<pre class="r"><code>Idents(seuratMfil) &lt;- seuratMfil$RNA_snn_res.0.25
DimPlot(seuratMfil, reduction = &quot;umap&quot;, pt.size = 0.1) </code></pre>
<p><img src="figure/CUP_knitting_easy_read.Rmd/data%20trimming-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##rerun seurat
seuratMfil &lt;- NormalizeData (object = seuratMfil)
seuratMfil &lt;- FindVariableFeatures(object = seuratMfil)
seuratMfil &lt;- ScaleData(object = seuratMfil, verbose = TRUE)
seuratMfil &lt;- RunPCA(object=seuratMfil, npcs = 30, verbose = FALSE)
seuratMfil &lt;- RunTSNE(object=seuratMfil, reduction=&quot;pca&quot;, dims = 1:20)
seuratMfil &lt;- RunUMAP(object=seuratMfil, reduction=&quot;pca&quot;, dims = 1:20)
seuratMfil &lt;- FindNeighbors(object = seuratMfil, reduction = &quot;pca&quot;, dims= 1:20)

res &lt;- c(0.25, 0.6, 0.8, 0.4)
for (i in 1:length(res)) {
  seuratMfil &lt;- FindClusters(object = seuratMfil, resolution = res[i], random.seed = 1234)
}</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 36040
Number of edges: 1257814

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9623
Number of communities: 13
Elapsed time: 5 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 36040
Number of edges: 1257814

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9334
Number of communities: 23
Elapsed time: 5 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 36040
Number of edges: 1257814

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9198
Number of communities: 28
Elapsed time: 4 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 36040
Number of edges: 1257814

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9495
Number of communities: 20
Elapsed time: 5 seconds</code></pre>
<div id="cup-non-hematopoietic-cells" class="section level3 tabset">
<h3 class="tabset">CUP non-hematopoietic cells</h3>
<div id="resolution-0.25" class="section level4">
<h4>Resolution 0.25</h4>
<pre class="r"><code>Idents(seuratMfil) &lt;- seuratMfil$RNA_snn_res.0.25
DimPlot(seuratMfil, reduction = &quot;umap&quot;, pt.size = 0.1) </code></pre>
<p><img src="figure/CUP_knitting_easy_read.Rmd/run%20UMAP%20plots_resolution-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="patient" class="section level4">
<h4>patient</h4>
<pre class="r"><code>Idents(seuratMfil) &lt;- seuratMfil$patient
DimPlot(seuratMfil, reduction = &quot;umap&quot;, pt.size = 0.1, shuffle = TRUE) </code></pre>
<p><img src="figure/CUP_knitting_easy_read.Rmd/run%20UMAP%20plots_patient-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="diseasecond" class="section level4">
<h4>diseaseCond</h4>
<pre class="r"><code>Idents(seuratMfil) &lt;- seuratMfil$diseaseCond
coldiseaseCond &lt;- c(&quot;#202547&quot;,&quot;lightgrey&quot;)
names(coldiseaseCond) &lt;- c(&quot;healthy&quot;, &quot;CUP&quot;)
DimPlot(seuratMfil, reduction = &quot;umap&quot;, pt.size = 0.1, cols = coldiseaseCond, order = &quot;healthy&quot;) </code></pre>
<p><img src="figure/CUP_knitting_easy_read.Rmd/run%20UMAP%20plots_diseaseCond-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="plot-features-cluster-marker" class="section level3">
<h3>plot features cluster marker</h3>
<pre class="r"><code>genelist &lt;- c(&quot;ENSG00000261371.PECAM1&quot;,&quot;ENSG00000117707.PROX1&quot;,&quot;ENSG00000163359.COL6A3&quot;,&quot;ENSG00000107796.ACTA2&quot;,&quot;ENSG00000164692.COL1A2&quot;,&quot;ENSG00000205420.KRT6A&quot;,&quot;ENSG00000119888.EPCAM&quot;,&quot;ENSG00000171345.KRT19&quot;,&quot;ENSG00000147889.CDKN2A&quot;)</code></pre>
</div>
<div id="marker-genes-from-distinct-cell-types"
class="section level3 tabset">
<h3 class="tabset">Marker genes from distinct cell types</h3>
<div id="endothelial-cells" class="section level4">
<h4>Endothelial cells</h4>
<pre class="r"><code>FeaturePlot(seuratMfil, reduction = &quot;umap&quot;, 
          features = genelist[1:2],
          cols=c(&quot;lightgrey&quot;, &quot;darkred&quot;),
          order = T )+
theme(legend.position=&quot;right&quot;, legend.title=element_text(size=3)) </code></pre>
<p><img src="figure/CUP_knitting_easy_read.Rmd/Endothelial%20cells-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="fibroblasts" class="section level4">
<h4>fibroblasts</h4>
<pre class="r"><code>FeaturePlot(seuratMfil, reduction = &quot;umap&quot;, 
          features = genelist[3:5],
          cols=c(&quot;lightgrey&quot;, &quot;darkred&quot;),
          order = T )+
theme(legend.position=&quot;right&quot;, legend.title=element_text(size=3)) </code></pre>
<p><img src="figure/CUP_knitting_easy_read.Rmd/fibroblasts-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="tumor-cells" class="section level4">
<h4>Tumor cells</h4>
<pre class="r"><code>FeaturePlot(seuratMfil, reduction = &quot;umap&quot;, 
          features = genelist[6:9],
          cols=c(&quot;lightgrey&quot;, &quot;darkred&quot;),
          order = T )+
theme(legend.position=&quot;right&quot;, legend.title=element_text(size=3)) </code></pre>
<p><img src="figure/CUP_knitting_easy_read.Rmd/Tumor%20cells-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##Endo
FeaturePlot(seuratMfil, features = &quot;ENSG00000261371.PECAM1&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))
FeaturePlot(seuratMfil, features = &quot;ENSG00000117707.PROX1&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))

##Fibroblastic
FeaturePlot(seuratMfil, features = &quot;ENSG00000163359.COL6A3&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))
FeaturePlot(seuratMfil, features = &quot;ENSG00000107796.ACTA2&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))
FeaturePlot(seuratMfil, features = &quot;ENSG00000164692.COL1A2&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))


##Tumor
FeaturePlot(seuratMfil, features = &quot;ENSG00000205420.KRT6A&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))
FeaturePlot(seuratMfil, features = &quot;ENSG00000119888.EPCAM&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))
FeaturePlot(seuratMfil, features = &quot;ENSG00000171345.KRT19&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))
FeaturePlot(seuratMfil, features = &quot;ENSG00000147889.CDKN2A&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
</div>
</div>
<div id="assign-cell-types-to-clusters" class="section level3">
<h3>Assign cell types to clusters</h3>
<pre class="r"><code>colclusterName &lt;- c(&quot;#67001f&quot;, &quot;#D53E4F&quot;, &quot;#f4a582&quot;, &quot;#FEE08B&quot;, &quot;#003c30&quot;,&quot;#01665e&quot;,&quot;#66C2A5&quot;, &quot;#3288BD&quot;,&quot;#BEAED4&quot;, &quot;#c7eae5&quot;,&quot;#355C7D&quot;,&quot;#202547&quot;,&quot;#B45B5C&quot;)
names(colclusterName) &lt;- c(&quot;FRC1&quot;,&quot;FRC2&quot;,&quot;Tu1&quot;,&quot;BEC1&quot;,&quot;CAF&quot;,&quot;BEC2&quot;, &quot;PRC&quot;,&quot;PRC_Tu&quot;,&quot;BEC3&quot;,&quot;Tu2&quot;,&quot;Tu3&quot;,&quot;LEC&quot;,&quot;PI16_FRC&quot;)


##assign celltypes
seuratMfil$colclusterName &lt;- &quot;celltype&quot;
seuratMfil$colclusterName[which(seuratMfil$RNA_snn_res.0.25 == &quot;0&quot;)] &lt;- &quot;FRC1&quot;
seuratMfil$colclusterName[which(seuratMfil$RNA_snn_res.0.25 == &quot;1&quot;)] &lt;- &quot;FRC2&quot;
seuratMfil$colclusterName[which(seuratMfil$RNA_snn_res.0.25 == &quot;2&quot;)] &lt;- &quot;Tu1&quot;
seuratMfil$colclusterName[which(seuratMfil$RNA_snn_res.0.25 == &quot;3&quot;)] &lt;- &quot;BEC1&quot;
seuratMfil$colclusterName[which(seuratMfil$RNA_snn_res.0.25 == &quot;4&quot;)] &lt;- &quot;CAF&quot;
seuratMfil$colclusterName[which(seuratMfil$RNA_snn_res.0.25 == &quot;5&quot;)] &lt;- &quot;BEC2&quot;
seuratMfil$colclusterName[which(seuratMfil$RNA_snn_res.0.25 == &quot;6&quot;)] &lt;- &quot;PRC&quot;
seuratMfil$colclusterName[which(seuratMfil$RNA_snn_res.0.25 == &quot;7&quot;)] &lt;- &quot;PRC_Tu&quot;
seuratMfil$colclusterName[which(seuratMfil$RNA_snn_res.0.25 == &quot;8&quot;)] &lt;- &quot;BEC3&quot;
seuratMfil$colclusterName[which(seuratMfil$RNA_snn_res.0.25 == &quot;9&quot;)] &lt;- &quot;Tu2&quot;
seuratMfil$colclusterName[which(seuratMfil$RNA_snn_res.0.25 == &quot;10&quot;)] &lt;- &quot;Tu3&quot;
seuratMfil$colclusterName[which(seuratMfil$RNA_snn_res.0.25 == &quot;11&quot;)] &lt;- &quot;LEC&quot;
seuratMfil$colclusterName[which(seuratMfil$RNA_snn_res.0.25 == &quot;12&quot;)] &lt;- &quot;PI16_FRC&quot;

table(seuratMfil$colclusterName)</code></pre>
<pre><code>
    BEC1     BEC2     BEC3      CAF     FRC1     FRC2      LEC PI16_FRC      PRC   PRC_Tu      Tu1 
    3606     2568     1885     3155     7823     4497      892      504     2565     2020     4377 
     Tu2      Tu3 
    1112     1036 </code></pre>
</div>
<div id="umap-with-defined-cluster-name" class="section level3 tabset">
<h3 class="tabset">UMAP with defined cluster name</h3>
<div id="umap-with-clustername" class="section level4">
<h4>UMAP with clusterName</h4>
<pre class="r"><code>Idents(seuratMfil) &lt;- seuratMfil$colclusterName
DimPlot(seuratMfil, reduction = &quot;umap&quot;, pt.size = 0.1, cols = colclusterName, label =TRUE) </code></pre>
<p><img src="figure/CUP_knitting_easy_read.Rmd/UMAP%20with%20clusterName-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="umap-with-clustername-split-by-disease" class="section level4">
<h4>UMAP with clusterName split by disease</h4>
<pre class="r"><code>Idents(seuratMfil) &lt;- seuratMfil$colclusterName
DimPlot(seuratMfil, reduction = &quot;umap&quot;, pt.size = 0.1, cols = colclusterName, label =FALSE, split.by = &quot;diseaseCond&quot;) </code></pre>
<p><img src="figure/CUP_knitting_easy_read.Rmd/UMAP%20with%20clusterName%20split%20by%20disease-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="umap-with-clustername-split-by-patients"
class="section level4">
<h4>UMAP with clusterName split by patients</h4>
<pre class="r"><code>Idents(seuratMfil) &lt;- seuratMfil$colclusterName
DimPlot(seuratMfil, reduction = &quot;umap&quot;, pt.size = 0.1, cols = colclusterName, label =FALSE, split.by = &quot;patient&quot;) </code></pre>
<p><img src="figure/CUP_knitting_easy_read.Rmd/UMAP%20with%20clusterName%20split%20by%20patients-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="calculate-cluster-marker-genes" class="section level3">
<h3>calculate cluster marker genes</h3>
<pre class="r"><code>##cluster marker
Idents(seuratMfil) &lt;- seuratMfil$colclusterName
markerGenes &lt;- FindAllMarkers(seuratMfil, only.pos=T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01)
#save table
write.table(markerGenes, 
            file= &quot;/R/CUP/analysis/markerGenes&quot;,
            sep=&quot;\t&quot;,
            quote=F,
            row.names=F,
            col.names=T)</code></pre>
</div>
<div id="plot-gene-signatures-onto-umap" class="section level3">
<h3>Plot gene signatures onto UMAP</h3>
<pre class="r"><code>##convert seurat object to sce object
sce &lt;- as.SingleCellExperiment(seuratMfil)


signatures &lt;- list(
  CAF = c(&quot;PDPN&quot;, &quot;POSTN&quot;, &quot;FAP&quot;, &quot;LRRC15&quot;),
  FRC = c(&quot;C7&quot;, &quot;CCL21&quot;,&quot;CCL19&quot;, &quot;CXCL12&quot;,&quot;IL33&quot;, &quot;PI16&quot;),
  PRC = c(&quot;ACTA2&quot;, &quot;ITGA7&quot;,&quot;RGS5&quot;, &quot;NOTCH3&quot;, &quot;MCAM&quot;),
  Tumor = c(&quot;KRT6A&quot;,&quot;EPCAM&quot;,&quot;KRT1&quot;,&quot;CDKN2A&quot;),
  Endothelial =c(&quot;PECAM1&quot;, &quot;VWF&quot;)
)


# Extract gene symbols
genes &lt;- data.frame(geneID = rownames(sce)) %&gt;%
  mutate(gene = gsub(&quot;.*\\.&quot;, &quot;&quot;, geneID))

# Function to compute signature and return UMAP plot
plot_signature_umap &lt;- function(sig_name, sig_genes) {
  signGenes &lt;- genes %&gt;% filter(gene %in% sig_genes)
  sceSub &lt;- sce[rownames(sce) %in% signGenes$geneID, ]
  
  cntMat &lt;- rowSums(t(as.matrix(sceSub@assays@data$logcounts))) / nrow(signGenes)
  sceSub$sign &lt;- cntMat
  sceSub$sign2 &lt;- pmin(sceSub$sign, 2.5)  # cap at 2.5
  
  plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 1) +
    scale_color_viridis_c() +  # or another color scale
    ggtitle(paste0(sig_name, &quot; signature&quot;)) +
    theme_minimal() +
    theme(legend.position = &quot;right&quot;)
}

# Generate all plots
umap_plots &lt;- lapply(names(signatures), function(name) {
  plot_signature_umap(name, signatures[[name]])
})</code></pre>
</div>
<div id="gene-signatures-of-cell-types" class="section level3 tabset">
<h3 class="tabset">Gene signatures of cell types</h3>
<div id="caf-gene-signatures" class="section level4">
<h4>CAF gene signatures</h4>
<pre class="r"><code>print(umap_plots[[1]])</code></pre>
<p><img src="figure/CUP_knitting_easy_read.Rmd/CAF%20gene%20signatures-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="frc-gene-signatures" class="section level4">
<h4>FRC gene signatures</h4>
<pre class="r"><code>print(umap_plots[[2]])</code></pre>
<p><img src="figure/CUP_knitting_easy_read.Rmd/FRC%20gene%20signatures-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="prc-gene-signatures" class="section level4">
<h4>PRC gene signatures</h4>
<pre class="r"><code>print(umap_plots[[3]])</code></pre>
<p><img src="figure/CUP_knitting_easy_read.Rmd/PRC%20gene%20signatures-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="tumor-gene-signatures" class="section level4">
<h4>Tumor gene signatures</h4>
<pre class="r"><code>print(umap_plots[[4]])</code></pre>
<p><img src="figure/CUP_knitting_easy_read.Rmd/Tumor%20gene%20signatures-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="endothelial-gene-signatures" class="section level4">
<h4>Endothelial gene signatures</h4>
<pre class="r"><code>print(umap_plots[[5]])</code></pre>
<p><img src="figure/CUP_knitting_easy_read.Rmd/Endothelial%20gene%20signatures-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="dotplot-marker-genes" class="section level3">
<h3>dotplot marker genes</h3>
<pre class="r"><code>seuratMfil$colclusterName &lt;- factor(seuratMfil$colclusterName, levels=c(&quot;PI16_FRC&quot;,&quot;FRC1&quot;,&quot;FRC2&quot;, &quot;PRC&quot;,&quot;PRC_Tu&quot;,&quot;CAF&quot;,&quot;Tu1&quot;,&quot;Tu2&quot;,&quot;Tu3&quot;,&quot;BEC1&quot;,&quot;BEC2&quot;,&quot;BEC3&quot;,&quot;LEC&quot;))
Idents(seuratMfil) &lt;-seuratMfil$colclusterName

genes &lt;- data.frame(gene=rownames(seuratMfil)) %&gt;% 
  mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))

selGenes &lt;- data.frame(geneID=rev(c( &quot;COL1A2&quot;, &quot;COL6A3&quot;, &quot;PDGFRB&quot;, &quot;DCN&quot;,  &quot;FBN1&quot;, &quot;C7&quot;, &quot;CCL21&quot;,&quot;CCL19&quot;, &quot;CXCL12&quot;,&quot;IL33&quot;, &quot;PI16&quot;, &quot;ACTA2&quot;, &quot;ITGA7&quot;,&quot;RGS5&quot;, &quot;NOTCH3&quot;, &quot;MCAM&quot;,&quot;PDPN&quot;, &quot;FAP&quot;,&quot;LRRC15&quot;,&quot;POSTN&quot;,&quot;PECAM1&quot;, &quot;VWF&quot;, &quot;PROX1&quot;))) %&gt;% left_join(., genes, by=&quot;geneID&quot;) %&gt;% filter(gene != &quot;ENSG00000232995.RGS5&quot;) 


DotPlot(seuratMfil, features = selGenes$gene, group.by = &quot;colclusterName&quot;) +
  RotatedAxis() +
  scale_color_viridis(option = &quot;F&quot;) +
  coord_flip()</code></pre>
<p><img src="figure/CUP_knitting_easy_read.Rmd/dotplot%20marker%20genes_total-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="relative-abundance-of-cell-types" class="section level3">
<h3>relative abundance of cell types</h3>
<pre class="r"><code>###dataset
datList &lt;- NULL
for(con in unique(seuratMfil$patient)){
  seuratSub &lt;- subset(seuratMfil, patient==con)
  print(dim(seuratSub))
  dat_con &lt;- as.data.frame(table(seuratSub$colclusterName)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(patient=con)
  datList[[con]] &lt;- dat_con
}</code></pre>
<pre><code>[1] 41620  5127
[1] 41620  2418
[1] 41620  5790
[1] 41620  5219
[1] 41620   118
[1] 41620  1296
[1] 41620  8582
[1] 41620  7490</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)


# Define the order you want
patient_order &lt;- c(&quot;hLN_01&quot;, &quot;hLN-02&quot;, &quot;hLN_03&quot;, &quot;hLN_04&quot;, &quot;hLN_05&quot;, &quot;CUP_01&quot;, &quot;CUP_03&quot;, &quot;CUP_04&quot;, &quot;CUP_05&quot;)  # customize as needed
dat_all$patient &lt;- factor(dat_all$patient, levels = patient_order)


## plot abundance - separated by patients
ggbarplot(dat_all, x= &quot;patient&quot;, y= &quot;percent&quot;, fill = &quot;Var1&quot;,  legend = &quot;right&quot;, legend.titel = &quot;cluster&quot;, ylab = &quot;frequency&quot;)  + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))</code></pre>
<p><img src="figure/CUP_knitting_easy_read.Rmd/rel%20abundance_total-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>###cond
###patient_diseaseCond
datList &lt;- NULL
for(con in unique(seuratMfil$diseaseCond)){
  seuratSub &lt;- subset(seuratMfil, diseaseCond==con)
  print(dim(seuratSub))
  dat_con &lt;- as.data.frame(table(seuratSub$colclusterName)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(diseaseCond=con)
  datList[[con]] &lt;- dat_con
}</code></pre>
<pre><code>[1] 41620 23617
[1] 41620 12423</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)


## plot abundance - separated by condition
ggbarplot(dat_all, x= &quot;diseaseCond&quot;, y= &quot;percent&quot;, fill = &quot;Var1&quot;, palette = colclusterName, legend = &quot;right&quot;, legend.titel = &quot;cluster&quot;, ylab = &quot;frequency&quot;)  + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) </code></pre>
<p><img src="figure/CUP_knitting_easy_read.Rmd/rel%20abundance_total-2.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="filter-the-object-with-fibroblasts-only"
class="section level3">
<h3>Filter the object with fibroblasts only</h3>
<pre class="r"><code>## Filter particular cell populations 
seuratM_fib_only &lt;- subset(seuratMfil, colclusterName %in% c(&quot;FRC1&quot;,&quot;FRC2&quot;,&quot;PRC&quot;,&quot;PI16_FRC&quot;,&quot;CAF&quot;,&quot;PRC_Tu&quot;)) 

colname_Fib &lt;- c(&quot;#67001f&quot;, &quot;#D53E4F&quot;, &quot;#003c30&quot;, &quot;#3288BD&quot;,&quot;#BEAED4&quot;, &quot;#c7eae5&quot;)
names(colname_Fib) &lt;- c(&quot;FRC1&quot;,&quot;FRC2&quot;,&quot;CAF&quot;,&quot;PRC&quot;,&quot;PRC_Tu&quot;,&quot;PI16_FRC&quot;)

##rerun seurat
seuratM_fib_only &lt;- NormalizeData (object = seuratM_fib_only)
seuratM_fib_only &lt;- FindVariableFeatures(object = seuratM_fib_only)
seuratM_fib_only &lt;- ScaleData(object = seuratM_fib_only, verbose = TRUE)
seuratM_fib_only &lt;- RunPCA(object=seuratM_fib_only, npcs = 30, verbose = FALSE)
seuratM_fib_only &lt;- RunTSNE(object=seuratM_fib_only, reduction=&quot;pca&quot;, dims = 1:20)
seuratM_fib_only &lt;- RunUMAP(object=seuratM_fib_only, reduction=&quot;pca&quot;, dims = 1:20)
seuratM_fib_only &lt;- FindNeighbors(object = seuratM_fib_only, reduction = &quot;pca&quot;, dims= 1:20)

res &lt;- c(0.25, 0.6, 0.8, 0.4)
for (i in 1:length(res)) {
  seuratM_fib_only &lt;- FindClusters(object = seuratM_fib_only, resolution = res[i], random.seed = 1234)
}</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 20564
Number of edges: 702640

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9422
Number of communities: 11
Elapsed time: 2 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 20564
Number of edges: 702640

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9086
Number of communities: 17
Elapsed time: 2 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 20564
Number of edges: 702640

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8936
Number of communities: 20
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 20564
Number of edges: 702640

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9265
Number of communities: 15
Elapsed time: 2 seconds</code></pre>
<pre class="r"><code>Idents(seuratM_fib_only) &lt;- seuratM_fib_only$colclusterName
DimPlot(seuratM_fib_only, reduction = &quot;umap&quot;, pt.size = 0.1) </code></pre>
<p><img src="figure/CUP_knitting_easy_read.Rmd/fibroblasts%20only-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seuratM_fib_only) &lt;- seuratM_fib_only$patient
DimPlot(seuratM_fib_only, reduction = &quot;umap&quot;, pt.size = 0.1, shuffle = TRUE) </code></pre>
<p><img src="figure/CUP_knitting_easy_read.Rmd/fibroblasts%20only-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seuratM_fib_only) &lt;- seuratM_fib_only$diseaseCond
coldiseaseCond &lt;- c(&quot;#202547&quot;,&quot;lightgrey&quot;)
names(coldiseaseCond) &lt;- c( &quot;CUP&quot;, &quot;healthy&quot;)
DimPlot(seuratM_fib_only, reduction = &quot;umap&quot;, pt.size = 0.1, cols = coldiseaseCond, order = &quot;CUP&quot;) </code></pre>
<p><img src="figure/CUP_knitting_easy_read.Rmd/fibroblasts%20only-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="relative-abundance-of-cell-types-1"
class="section level3 tabset">
<h3 class="tabset">relative abundance of cell types</h3>
<div id="rel-abundance_fibroblasts-spilt-by-patients"
class="section level4">
<h4>rel abundance_fibroblasts spilt by patients</h4>
<pre class="r"><code>datList &lt;- NULL
for(con in unique(seuratM_fib_only$patient)){
  seuratSub &lt;- subset(seuratM_fib_only, patient==con)
  print(dim(seuratSub))
  dat_con &lt;- as.data.frame(table(seuratSub$colclusterName)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(patient=con)
  datList[[con]] &lt;- dat_con
}</code></pre>
<pre><code>[1] 41620  5121
[1] 41620  2292
[1] 41620  1179
[1] 41620  3844
[1] 41620    54
[1] 41620   121
[1] 41620  4549
[1] 41620  3404</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)

# Define the order you want
patient_order &lt;- c(&quot;hLN_01&quot;, &quot;hLN-02&quot;, &quot;hLN_03&quot;, &quot;hLN_04&quot;, &quot;hLN_05&quot;, &quot;CUP_01&quot;, &quot;CUP_03&quot;, &quot;CUP_04&quot;, &quot;CUP_05&quot;)  # customize as needed
dat_all$patient &lt;- factor(dat_all$patient, levels = patient_order)

# plot abundance - separated by patients
ggbarplot(dat_all, x= &quot;patient&quot;, y= &quot;percent&quot;, fill = &quot;Var1&quot;, palette = colname_Fib,  legend = &quot;right&quot;, legend.titel = &quot;cluster&quot;, ylab = &quot;frequency&quot;)  + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))</code></pre>
<p><img src="figure/CUP_knitting_easy_read.Rmd/rel%20abundance_fibroblasts%20spilt%20by%20patients-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="rel-abundance_fibroblasts-spilt-by-condition"
class="section level4">
<h4>rel abundance_fibroblasts spilt by condition</h4>
<pre class="r"><code>datList &lt;- NULL
for(con in unique(seuratM_fib_only$diseaseCond)){
  seuratSub &lt;- subset(seuratM_fib_only, diseaseCond==con)
  print(dim(seuratSub))
  dat_con &lt;- as.data.frame(table(seuratSub$colclusterName)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(diseaseCond=con)
  datList[[con]] &lt;- dat_con
}</code></pre>
<pre><code>[1] 41620 15366
[1] 41620  5198</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)

# plot abundance - separated by condition
ggbarplot(dat_all, x= &quot;diseaseCond&quot;, y= &quot;percent&quot;, fill = &quot;Var1&quot;, palette = colname_Fib, legend = &quot;right&quot;, legend.titel = &quot;cluster&quot;, ylab = &quot;frequency&quot;)  + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) </code></pre>
<p><img src="figure/CUP_knitting_easy_read.Rmd/rel%20abundance_fibroblasts%20spilt%20by%20condition-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="session-info" class="section level2">
<h2>session info</h2>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Fri May  2 12:58:06 2025&quot;</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 11 x64 (build 22631)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8  LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] NCmisc_1.2.0                VennDiagram_1.7.3           futile.logger_1.4.3        
 [4] ggupset_0.4.0               gridExtra_2.3               DOSE_3.30.5                
 [7] enrichplot_1.24.4           msigdbr_7.5.1               org.Hs.eg.db_3.19.1        
[10] AnnotationDbi_1.66.0        clusterProfiler_4.12.6      multtest_2.60.0            
[13] metap_1.11                  scater_1.32.1               scuttle_1.14.0             
[16] destiny_3.18.0              circlize_0.4.16             muscat_1.18.0              
[19] viridis_0.6.5               viridisLite_0.4.2           lubridate_1.9.3            
[22] forcats_1.0.0               stringr_1.5.1               purrr_1.0.2                
[25] readr_2.1.5                 tidyr_1.3.1                 tibble_3.2.1               
[28] tidyverse_2.0.0             dplyr_1.1.4                 SingleCellExperiment_1.26.0
[31] SummarizedExperiment_1.34.0 Biobase_2.64.0              GenomicRanges_1.56.1       
[34] GenomeInfoDb_1.40.1         IRanges_2.38.1              S4Vectors_0.42.1           
[37] BiocGenerics_0.50.0         MatrixGenerics_1.16.0       matrixStats_1.4.1          
[40] pheatmap_1.0.12             ggpubr_0.6.0                ggplot2_3.5.1              
[43] Seurat_5.1.0                SeuratObject_5.0.2          sp_2.1-4                   
[46] runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] igraph_2.0.3              ica_1.0-3                 plotly_4.10.4            
  [4] Formula_1.2-5             zlibbioc_1.50.0           tidyselect_1.2.1         
  [7] bit_4.0.5                 doParallel_1.0.17         clue_0.3-65              
 [10] lattice_0.22-6            rjson_0.2.23              blob_1.2.4               
 [13] S4Arrays_1.4.1            pbkrtest_0.5.3            parallel_4.4.1           
 [16] png_0.1-8                 plotrix_3.8-4             cli_3.6.3                
 [19] ggplotify_0.1.2           goftest_1.2-3             VIM_6.2.2                
 [22] variancePartition_1.34.0  BiocNeighbors_1.22.0      shadowtext_0.1.4         
 [25] uwot_0.2.2                curl_5.2.3                tidytree_0.4.6           
 [28] mime_0.12                 evaluate_1.0.0            leiden_0.4.3.1           
 [31] ComplexHeatmap_2.20.0     stringi_1.8.4             backports_1.5.0          
 [34] lmerTest_3.1-3            qqconf_1.3.2              httpuv_1.6.15            
 [37] magrittr_2.0.3            rappdirs_0.3.3            splines_4.4.1            
 [40] ggraph_2.2.1              sctransform_0.4.1         ggbeeswarm_0.7.2         
 [43] DBI_1.2.3                 jquerylib_0.1.4           smoother_1.3             
 [46] withr_3.0.1               git2r_0.33.0              corpcor_1.6.10           
 [49] reformulas_0.3.0          class_7.3-22              rprojroot_2.0.4          
 [52] lmtest_0.9-40             tidygraph_1.3.1           formatR_1.14             
 [55] colourpicker_1.3.0        htmlwidgets_1.6.4         fs_1.6.4                 
 [58] ggrepel_0.9.6             labeling_0.4.3            fANCOVA_0.6-1            
 [61] SparseArray_1.4.8         DESeq2_1.44.0             ranger_0.16.0            
 [64] DEoptimR_1.1-3            reticulate_1.39.0         hexbin_1.28.4            
 [67] zoo_1.8-12                XVector_0.44.0            knitr_1.48               
 [70] ggplot.multistats_1.0.1   UCSC.utils_1.0.0          RhpcBLASctl_0.23-42      
 [73] timechange_0.3.0          foreach_1.5.2             fansi_1.0.6              
 [76] patchwork_1.3.0           caTools_1.18.3            ggtree_3.12.0            
 [79] data.table_1.15.4         R.oo_1.26.0               RSpectra_0.16-2          
 [82] irlba_2.3.5.1             gridGraphics_0.5-1        fastDummies_1.7.4        
 [85] lazyeval_0.2.2            yaml_2.3.10               survival_3.6-4           
 [88] scattermore_1.2           crayon_1.5.3              RcppAnnoy_0.0.22         
 [91] RColorBrewer_1.1-3        progressr_0.14.0          tweenr_2.0.3             
 [94] later_1.3.2               ggridges_0.5.6            codetools_0.2-20         
 [97] GlobalOptions_0.1.2       aod_1.3.3                 KEGGREST_1.44.1          
[100] Rtsne_0.17                shape_1.4.6.1             limma_3.60.4             
[103] pkgconfig_2.0.3           TMB_1.9.15                spatstat.univar_3.0-1    
[106] mathjaxr_1.6-0            EnvStats_3.0.0            aplot_0.2.3              
[109] scatterplot3d_0.3-44      ape_5.8                   spatstat.sparse_3.1-0    
[112] xtable_1.8-4              car_3.1-3                 highr_0.11               
[115] plyr_1.8.9                httr_1.4.7                rbibutils_2.2.16         
[118] tools_4.4.1               globals_0.16.3            beeswarm_0.4.0           
[121] broom_1.0.7               nlme_3.1-164              lambda.r_1.2.4           
[124] lme4_1.1-35.5             digest_0.6.36             numDeriv_2016.8-1.1      
[127] Matrix_1.7-0              farver_2.1.2              tzdb_0.4.0               
[130] remaCor_0.0.18            reshape2_1.4.4            yulab.utils_0.1.7        
[133] glue_1.7.0                cachem_1.1.0              polyclip_1.10-7          
[136] generics_0.1.3            Biostrings_2.72.1         mvtnorm_1.3-1            
[139] parallelly_1.38.0         mnormt_2.1.1              statmod_1.5.0            
[142] RcppHNSW_0.6.0            ScaledMatrix_1.12.0       carData_3.0-5            
[145] minqa_1.2.8               pbapply_1.7-2             httr2_1.0.5              
[148] spam_2.10-0               gson_0.1.0                utf8_1.2.4               
[151] graphlayouts_1.2.0        gtools_3.9.5              ggsignif_0.6.4           
[154] RcppEigen_0.3.4.0.2       shiny_1.9.1               GenomeInfoDbData_1.2.12  
[157] glmmTMB_1.1.10            R.utils_2.12.3            memoise_2.0.1            
[160] rmarkdown_2.28            scales_1.3.0              R.methodsS3_1.8.2        
[163] future_1.34.0             RANN_2.6.1                Cairo_1.6-2              
[166] spatstat.data_3.1-2       rstudioapi_0.16.0         cluster_2.1.6            
[169] mutoss_0.1-13             spatstat.utils_3.1-0      hms_1.1.3                
[172] fitdistrplus_1.2-1        munsell_0.5.1             cowplot_1.1.3            
[175] colorspace_2.1-0          rlang_1.1.4               DelayedMatrixStats_1.26.0
[178] sparseMatrixStats_1.16.0  xts_0.14.0                dotCall64_1.1-1          
[181] shinydashboard_0.7.2      ggforce_0.4.2             laeken_0.5.3             
[184] mgcv_1.9-1                xfun_0.46                 e1071_1.7-16             
[187] TH.data_1.1-2             iterators_1.0.14          abind_1.4-8              
[190] GOSemSim_2.30.2           treeio_1.28.0             futile.options_1.0.1     
[193] bitops_1.0-8              Rdpack_2.6.1              promises_1.3.0           
[196] scatterpie_0.2.4          RSQLite_2.3.7             qvalue_2.36.0            
[199] sandwich_3.1-1            fgsea_1.30.0              DelayedArray_0.30.1      
[202] proxy_0.4-27              GO.db_3.19.1              compiler_4.4.1           
[205] prettyunits_1.2.0         boot_1.3-30               beachmat_2.20.0          
[208] listenv_0.9.1             Rcpp_1.0.13               edgeR_4.2.1              
[211] workflowr_1.7.1           BiocSingular_1.20.0       tensor_1.5               
[214] MASS_7.3-60.2             progress_1.2.3            BiocParallel_1.38.0      
[217] babelgene_22.9            spatstat.random_3.3-1     R6_2.5.1                 
[220] fastmap_1.2.0             multcomp_1.4-26           fastmatch_1.1-4          
[223] rstatix_0.7.2             vipor_0.4.7               TTR_0.24.4               
[226] ROCR_1.0-11               TFisher_0.2.0             rsvd_1.0.5               
[229] vcd_1.4-13                nnet_7.3-19               gtable_0.3.5             
[232] KernSmooth_2.23-24        miniUI_0.1.1.1            deldir_2.0-4             
[235] htmltools_0.5.8.1         ggthemes_5.1.0            bit64_4.0.5              
[238] spatstat.explore_3.3-2    lifecycle_1.0.4           blme_1.0-6               
[241] nloptr_2.1.1              sass_0.4.9                vctrs_0.6.5              
[244] robustbase_0.99-4         spatstat.geom_3.3-2       sn_2.1.1                 
[247] ggfun_0.1.6               future.apply_1.11.2       bslib_0.8.0              
[250] pillar_1.9.0              gplots_3.1.3.1            pcaMethods_1.96.0        
[253] locfit_1.5-9.10           jsonlite_1.8.8            GetoptLong_1.0.5         </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 11 x64 (build 22631)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8  LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] NCmisc_1.2.0                VennDiagram_1.7.3           futile.logger_1.4.3        
 [4] ggupset_0.4.0               gridExtra_2.3               DOSE_3.30.5                
 [7] enrichplot_1.24.4           msigdbr_7.5.1               org.Hs.eg.db_3.19.1        
[10] AnnotationDbi_1.66.0        clusterProfiler_4.12.6      multtest_2.60.0            
[13] metap_1.11                  scater_1.32.1               scuttle_1.14.0             
[16] destiny_3.18.0              circlize_0.4.16             muscat_1.18.0              
[19] viridis_0.6.5               viridisLite_0.4.2           lubridate_1.9.3            
[22] forcats_1.0.0               stringr_1.5.1               purrr_1.0.2                
[25] readr_2.1.5                 tidyr_1.3.1                 tibble_3.2.1               
[28] tidyverse_2.0.0             dplyr_1.1.4                 SingleCellExperiment_1.26.0
[31] SummarizedExperiment_1.34.0 Biobase_2.64.0              GenomicRanges_1.56.1       
[34] GenomeInfoDb_1.40.1         IRanges_2.38.1              S4Vectors_0.42.1           
[37] BiocGenerics_0.50.0         MatrixGenerics_1.16.0       matrixStats_1.4.1          
[40] pheatmap_1.0.12             ggpubr_0.6.0                ggplot2_3.5.1              
[43] Seurat_5.1.0                SeuratObject_5.0.2          sp_2.1-4                   
[46] runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] igraph_2.0.3              ica_1.0-3                 plotly_4.10.4            
  [4] Formula_1.2-5             zlibbioc_1.50.0           tidyselect_1.2.1         
  [7] bit_4.0.5                 doParallel_1.0.17         clue_0.3-65              
 [10] lattice_0.22-6            rjson_0.2.23              blob_1.2.4               
 [13] S4Arrays_1.4.1            pbkrtest_0.5.3            parallel_4.4.1           
 [16] png_0.1-8                 plotrix_3.8-4             cli_3.6.3                
 [19] ggplotify_0.1.2           goftest_1.2-3             VIM_6.2.2                
 [22] variancePartition_1.34.0  BiocNeighbors_1.22.0      shadowtext_0.1.4         
 [25] uwot_0.2.2                curl_5.2.3                tidytree_0.4.6           
 [28] mime_0.12                 evaluate_1.0.0            leiden_0.4.3.1           
 [31] ComplexHeatmap_2.20.0     stringi_1.8.4             backports_1.5.0          
 [34] lmerTest_3.1-3            qqconf_1.3.2              httpuv_1.6.15            
 [37] magrittr_2.0.3            rappdirs_0.3.3            splines_4.4.1            
 [40] ggraph_2.2.1              sctransform_0.4.1         ggbeeswarm_0.7.2         
 [43] DBI_1.2.3                 jquerylib_0.1.4           smoother_1.3             
 [46] withr_3.0.1               git2r_0.33.0              corpcor_1.6.10           
 [49] reformulas_0.3.0          class_7.3-22              rprojroot_2.0.4          
 [52] lmtest_0.9-40             tidygraph_1.3.1           formatR_1.14             
 [55] colourpicker_1.3.0        htmlwidgets_1.6.4         fs_1.6.4                 
 [58] ggrepel_0.9.6             labeling_0.4.3            fANCOVA_0.6-1            
 [61] SparseArray_1.4.8         DESeq2_1.44.0             ranger_0.16.0            
 [64] DEoptimR_1.1-3            reticulate_1.39.0         hexbin_1.28.4            
 [67] zoo_1.8-12                XVector_0.44.0            knitr_1.48               
 [70] ggplot.multistats_1.0.1   UCSC.utils_1.0.0          RhpcBLASctl_0.23-42      
 [73] timechange_0.3.0          foreach_1.5.2             fansi_1.0.6              
 [76] patchwork_1.3.0           caTools_1.18.3            ggtree_3.12.0            
 [79] data.table_1.15.4         R.oo_1.26.0               RSpectra_0.16-2          
 [82] irlba_2.3.5.1             gridGraphics_0.5-1        fastDummies_1.7.4        
 [85] lazyeval_0.2.2            yaml_2.3.10               survival_3.6-4           
 [88] scattermore_1.2           crayon_1.5.3              RcppAnnoy_0.0.22         
 [91] RColorBrewer_1.1-3        progressr_0.14.0          tweenr_2.0.3             
 [94] later_1.3.2               ggridges_0.5.6            codetools_0.2-20         
 [97] GlobalOptions_0.1.2       aod_1.3.3                 KEGGREST_1.44.1          
[100] Rtsne_0.17                shape_1.4.6.1             limma_3.60.4             
[103] pkgconfig_2.0.3           TMB_1.9.15                spatstat.univar_3.0-1    
[106] mathjaxr_1.6-0            EnvStats_3.0.0            aplot_0.2.3              
[109] scatterplot3d_0.3-44      ape_5.8                   spatstat.sparse_3.1-0    
[112] xtable_1.8-4              car_3.1-3                 highr_0.11               
[115] plyr_1.8.9                httr_1.4.7                rbibutils_2.2.16         
[118] tools_4.4.1               globals_0.16.3            beeswarm_0.4.0           
[121] broom_1.0.7               nlme_3.1-164              lambda.r_1.2.4           
[124] lme4_1.1-35.5             digest_0.6.36             numDeriv_2016.8-1.1      
[127] Matrix_1.7-0              farver_2.1.2              tzdb_0.4.0               
[130] remaCor_0.0.18            reshape2_1.4.4            yulab.utils_0.1.7        
[133] glue_1.7.0                cachem_1.1.0              polyclip_1.10-7          
[136] generics_0.1.3            Biostrings_2.72.1         mvtnorm_1.3-1            
[139] parallelly_1.38.0         mnormt_2.1.1              statmod_1.5.0            
[142] RcppHNSW_0.6.0            ScaledMatrix_1.12.0       carData_3.0-5            
[145] minqa_1.2.8               pbapply_1.7-2             httr2_1.0.5              
[148] spam_2.10-0               gson_0.1.0                utf8_1.2.4               
[151] graphlayouts_1.2.0        gtools_3.9.5              ggsignif_0.6.4           
[154] RcppEigen_0.3.4.0.2       shiny_1.9.1               GenomeInfoDbData_1.2.12  
[157] glmmTMB_1.1.10            R.utils_2.12.3            memoise_2.0.1            
[160] rmarkdown_2.28            scales_1.3.0              R.methodsS3_1.8.2        
[163] future_1.34.0             RANN_2.6.1                Cairo_1.6-2              
[166] spatstat.data_3.1-2       rstudioapi_0.16.0         cluster_2.1.6            
[169] mutoss_0.1-13             spatstat.utils_3.1-0      hms_1.1.3                
[172] fitdistrplus_1.2-1        munsell_0.5.1             cowplot_1.1.3            
[175] colorspace_2.1-0          rlang_1.1.4               DelayedMatrixStats_1.26.0
[178] sparseMatrixStats_1.16.0  xts_0.14.0                dotCall64_1.1-1          
[181] shinydashboard_0.7.2      ggforce_0.4.2             laeken_0.5.3             
[184] mgcv_1.9-1                xfun_0.46                 e1071_1.7-16             
[187] TH.data_1.1-2             iterators_1.0.14          abind_1.4-8              
[190] GOSemSim_2.30.2           treeio_1.28.0             futile.options_1.0.1     
[193] bitops_1.0-8              Rdpack_2.6.1              promises_1.3.0           
[196] scatterpie_0.2.4          RSQLite_2.3.7             qvalue_2.36.0            
[199] sandwich_3.1-1            fgsea_1.30.0              DelayedArray_0.30.1      
[202] proxy_0.4-27              GO.db_3.19.1              compiler_4.4.1           
[205] prettyunits_1.2.0         boot_1.3-30               beachmat_2.20.0          
[208] listenv_0.9.1             Rcpp_1.0.13               edgeR_4.2.1              
[211] workflowr_1.7.1           BiocSingular_1.20.0       tensor_1.5               
[214] MASS_7.3-60.2             progress_1.2.3            BiocParallel_1.38.0      
[217] babelgene_22.9            spatstat.random_3.3-1     R6_2.5.1                 
[220] fastmap_1.2.0             multcomp_1.4-26           fastmatch_1.1-4          
[223] rstatix_0.7.2             vipor_0.4.7               TTR_0.24.4               
[226] ROCR_1.0-11               TFisher_0.2.0             rsvd_1.0.5               
[229] vcd_1.4-13                nnet_7.3-19               gtable_0.3.5             
[232] KernSmooth_2.23-24        miniUI_0.1.1.1            deldir_2.0-4             
[235] htmltools_0.5.8.1         ggthemes_5.1.0            bit64_4.0.5              
[238] spatstat.explore_3.3-2    lifecycle_1.0.4           blme_1.0-6               
[241] nloptr_2.1.1              sass_0.4.9                vctrs_0.6.5              
[244] robustbase_0.99-4         spatstat.geom_3.3-2       sn_2.1.1                 
[247] ggfun_0.1.6               future.apply_1.11.2       bslib_0.8.0              
[250] pillar_1.9.0              gplots_3.1.3.1            pcaMethods_1.96.0        
[253] locfit_1.5-9.10           jsonlite_1.8.8            GetoptLong_1.0.5         </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
