<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="HW.Cheng" />

<meta name="date" content="2025-04-25" />

<title>CUP Umaps</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CUP</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">CUP Umaps</h1>
<h4 class="author">HW.Cheng</h4>
<h4 class="date">2025-04-25</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-07-22
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>CUP/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250318code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250318)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250318code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250318)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/R/CUP/data/CD45pos/
</td>
<td style="text-align:left;">
data/CD45pos
</td>
</tr>
<tr>
<td style="text-align:left;">
/R/CUP/data/CD45pos_seurat_merge.rds
</td>
<td style="text-align:left;">
data/CD45pos_seurat_merge.rds
</td>
</tr>
<tr>
<td style="text-align:left;">
/R/CUP/analysis/markerGenes_CD45pos
</td>
<td style="text-align:left;">
analysis/markerGenes_CD45pos
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomHWC0406CUPtreee1bdb0b292ad4fdded81e5b38224bf7970f1063etargetblanke1bdb0ba">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/HWC0406/CUP/tree/e1bdb0b292ad4fdded81e5b38224bf7970f1063e" target="_blank">e1bdb0b</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomHWC0406CUPtreee1bdb0b292ad4fdded81e5b38224bf7970f1063etargetblanke1bdb0ba"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/HWC0406/CUP/tree/e1bdb0b292ad4fdded81e5b38224bf7970f1063e" target="_blank">e1bdb0b</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/figure/
    Ignored:    data/CD45pos/
    Ignored:    data/seurat_objects/

Untracked files:
    Untracked:  analysis/CUP_CD45_merge_healthy.Rmd
    Untracked:  analysis/CUP_knitting_easy_read.Rmd
    Untracked:  analysis/markerGenes_CD45pos

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="r-markdown" class="section level2">
<h2>R Markdown</h2>
<p>This is an R Markdown document. Markdown is a simple formatting
syntax for authoring HTML, PDF, and MS Word documents. For more details
on using R Markdown see <a href="http://rmarkdown.rstudio.com"
class="uri">http://rmarkdown.rstudio.com</a>.</p>
<p>When you click the <strong>Knit</strong> button a document will be
generated that includes both content as well as the output of any
embedded R code chunks within the document. You can embed an R code
chunk like this:</p>
</div>
<div id="load-packages" class="section level2">
<h2>load packages</h2>
</div>
<div id="load-files-and-merge" class="section level2">
<h2>load files and merge</h2>
<pre class="r"><code>##load merged file 
fileNam &lt;- &quot;/R/CUP/data/CD45pos_seurat_merge.rds&quot;
seuratM &lt;- readRDS(fileNam)
table(seuratM$dataset)</code></pre>
<pre><code>
 373291_02-2_20241211_Hu_CUP_01_CD45pos  373291_04-4_20241211_Hu_CUP_04_CD45pos 
                                   8611                                    5382 
 373291_06-6_20241211_Hu_CUP_05_CD45pos  373291_08-8_20241211_Hu_CUP_03_CD45pos 
                                   6703                                    6774 
    o27533_1_02-2_20220201_Hu_LN_CD3_v3    o27533_1_03-3_20220201_Hu_LN_CD19_v3 
                                  10734                                    7665 
o27533_1_04-4_20220201_Hu_LN_Myeloid_v3     o27533_1_06-6_20220202_Hu_LN_CD3_v3 
                                   6182                                   12724 
   o27533_1_07-7_20220202_Hu_LN_CD19_v3 o27533_1_08-8_20220202_Hu_LN_Myeloid_v3 
                                   7525                                   10287 </code></pre>
<pre class="r"><code>table(seuratM$RNA_snn_res.0.25)</code></pre>
<pre><code>
    0     1     2     3     4     5     6     7     8     9    10    11    12    13    14    15 
21705  9651  8382  7672  7020  5968  5164  2988  2824  2525  2059  1696  1662  1140   709   469 
   16    17    18    19 
  326   262   183   182 </code></pre>
<pre class="r"><code>table(seuratM$orig.ident)</code></pre>
<pre><code>
      
82587 </code></pre>
</div>
<div id="add-metadata" class="section level2">
<h2>add metadata</h2>
<pre class="r"><code>### add any type of metadata

CUP_pat_01 &lt;- c(&quot;373291_02-2_20241211_Hu_CUP_01_CD45pos&quot;)
CUP_pat_04 &lt;- c(&quot;373291_04-4_20241211_Hu_CUP_04_CD45pos&quot;)
CUP_pat_05 &lt;- c(&quot;373291_06-6_20241211_Hu_CUP_05_CD45pos&quot;)
CUP_pat_03 &lt;- c(&quot;373291_08-8_20241211_Hu_CUP_03_CD45pos&quot;)
hLN_04 &lt;- c(&quot;o27533_1_02-2_20220201_Hu_LN_CD3_v3&quot;,&quot;o27533_1_03-3_20220201_Hu_LN_CD19_v3&quot;,&quot;o27533_1_04-4_20220201_Hu_LN_Myeloid_v3&quot;)
hLN_05 &lt;- c(&quot;o27533_1_06-6_20220202_Hu_LN_CD3_v3&quot;,&quot;o27533_1_07-7_20220202_Hu_LN_CD19_v3&quot;,&quot;o27533_1_08-8_20220202_Hu_LN_Myeloid_v3&quot;)

seuratM$patient &lt;- &quot;pat_nr&quot;
seuratM$patient[which(seuratM$dataset %in% CUP_pat_01)] &lt;- &quot;CUP_01&quot;
seuratM$patient[which(seuratM$dataset %in% CUP_pat_04)] &lt;- &quot;CUP_04&quot;
seuratM$patient[which(seuratM$dataset %in% CUP_pat_05)] &lt;- &quot;CUP_05&quot;
seuratM$patient[which(seuratM$dataset %in% CUP_pat_03)] &lt;- &quot;CUP_03&quot;
seuratM$patient[which(seuratM$dataset %in% hLN_04)] &lt;- &quot;hLN_04&quot;
seuratM$patient[which(seuratM$dataset %in% hLN_05)] &lt;- &quot;hLN_05&quot;

table(seuratM$patient)</code></pre>
<pre><code>
CUP_01 CUP_03 CUP_04 CUP_05 hLN_04 hLN_05 
  8611   6774   5382   6703  24581  30536 </code></pre>
<pre class="r"><code>table(seuratM$orig.ident)</code></pre>
<pre><code>
      
82587 </code></pre>
<pre class="r"><code>#### diseaseCond
seuratM$diseaseCond &lt;- &quot;diseaseCond&quot;
seuratM$diseaseCond[grepl(&quot;CUP&quot;, seuratM$dataset)] &lt;- &quot;CUP&quot;
seuratM$diseaseCond[grepl(&quot;LN&quot;, seuratM$dataset)] &lt;- &quot;healthy&quot;
table(seuratM$diseaseCond)</code></pre>
<pre><code>
    CUP healthy 
  27470   55117 </code></pre>
<pre class="r"><code>###order
orddiseaseCond &lt;- c(&quot;healthy&quot;, &quot;CUP&quot;)
seuratM$diseaseCond &lt;- factor(seuratM$diseaseCond, levels=c(&quot;healthy&quot;, &quot;CUP&quot;))
Idents(seuratM) &lt;- seuratM$diseaseCond
table(seuratM$diseaseCond)</code></pre>
<pre><code>
healthy     CUP 
  55117   27470 </code></pre>
</div>
<div id="rerun-seurat" class="section level2">
<h2>rerun seurat</h2>
<pre class="r"><code>#rerun seurat
seuratM &lt;- NormalizeData (object = seuratM)
seuratM &lt;- FindVariableFeatures(object = seuratM, selection.method = &quot;vst&quot;, nfeatures = 3000)
seuratM &lt;- ScaleData(object = seuratM, verbose = TRUE)
seuratM &lt;- RunPCA(object=seuratM, npcs = 20, verbose = FALSE)
seuratM &lt;- RunTSNE(object=seuratM, reduction=&quot;pca&quot;, dims = 1:20, check_duplicates = FALSE)
seuratM &lt;- RunUMAP(object=seuratM, reduction=&quot;pca&quot;, dims = 1:20)
seuratM &lt;- FindNeighbors(object = seuratM, reduction = &quot;pca&quot;, dims= 1:20)

res &lt;- c(0.25, 0.6, 0.8, 0.4, 0.1)
for (i in 1:length(res)) {
  seuratM &lt;- FindClusters(object = seuratM, resolution = res[i], random.seed = 1234)
}</code></pre>
</div>
<div id="plot-cell-numbers" class="section level2">
<h2>plot cell numbers</h2>
<pre class="r"><code>cell_count &lt;- data.frame(table(seuratM$patient))
colnames(cell_count) &lt;- c(&quot;patient&quot;, &quot;Freq&quot;)
hsize &lt;- 1.5

ggplot(cell_count, aes(x = hsize, y = Freq, fill = patient)) +
  #scale_fill_manual(values = colpat2) +
  geom_col(color = &quot;white&quot;) +
  coord_polar(theta = &quot;y&quot;) +
  xlim(c(0.2, hsize + 0.5)) +
  theme_void() +
  ggtitle(&quot;cell number&quot;) +
  theme(plot.title = element_text(hjust = 0.5, size = 15)) +
  geom_text(aes(label = Freq), position = position_stack(vjust = 0.5))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/cellnumber%20circle-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="plot-umaps" class="section level2">
<h2>plot umaps</h2>
<pre class="r"><code>Idents(seuratM) &lt;- seuratM$RNA_snn_res.0.1
DimPlot(seuratM, reduction = &quot;umap&quot;, pt.size = 0.1,label = TRUE) </code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/umap-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seuratM) &lt;- seuratM$patient
DimPlot(seuratM, reduction = &quot;umap&quot;, pt.size = 0.1, shuffle = TRUE) </code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/umap-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seuratM) &lt;- seuratM$diseaseCond
coldiseaseCond &lt;- c(&quot;#202547&quot;,&quot;lightgrey&quot;)
names(coldiseaseCond) &lt;- c(&quot;healthy&quot;, &quot;CUP&quot;)
DimPlot(seuratM, reduction = &quot;umap&quot;, pt.size = 0.1, cols = coldiseaseCond, order = &quot;healthy&quot;) </code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/umap-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="calculate-cluster-marker-genes" class="section level2">
<h2>calculate cluster marker genes</h2>
<pre class="r"><code>##cluster marker
Idents(seuratM) &lt;- seuratM$RNA_snn_res.0.1
markerGenes &lt;- FindAllMarkers(seuratM, only.pos=T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01)
#save table
write.table(markerGenes, 
            file= &quot;/R/CUP/analysis/markerGenes_CD45pos&quot;,
            sep=&quot;\t&quot;,
            quote=F,
            row.names=F,
            col.names=T)</code></pre>
</div>
<div id="assign-cell-types-to-clusters" class="section level2">
<h2>Assign cell types to clusters</h2>
<pre class="r"><code>colclusterName &lt;- c(&quot;#67001f&quot;, &quot;#D53E4F&quot;, &quot;#f4a582&quot;, &quot;#FEE08B&quot;,&quot;#66C2A5&quot;, &quot;#3288BD&quot;,&quot;#BEAED4&quot;, &quot;#c7eae5&quot;,&quot;#355C7D&quot;,&quot;#202547&quot;,&quot;#B45B5C&quot;)
names(colclusterName) &lt;- c(&quot;T_cell_1&quot;,&quot;T_cell_2&quot;,&quot;B_cell_1&quot;,&quot;B_cell_2&quot;,&quot;B_cell_3&quot;,&quot;NK&quot;, &quot;Mph1&quot;,&quot;Mph2&quot;,&quot;proliferating&quot;,&quot;pDC_1&quot;,&quot;pDC_2&quot;)


##assign celltypes
seuratM$colclusterName &lt;- &quot;celltype&quot;
seuratM$colclusterName[which(seuratM$RNA_snn_res.0.1 == &quot;0&quot;)] &lt;- &quot;T_cell_1&quot;
seuratM$colclusterName[which(seuratM$RNA_snn_res.0.1 == &quot;1&quot;)] &lt;- &quot;T_cell_2&quot;
seuratM$colclusterName[which(seuratM$RNA_snn_res.0.1 == &quot;2&quot;)] &lt;- &quot;B_cell_1&quot;
seuratM$colclusterName[which(seuratM$RNA_snn_res.0.1 == &quot;3&quot;)] &lt;- &quot;B_cell_2&quot;
seuratM$colclusterName[which(seuratM$RNA_snn_res.0.1 == &quot;4&quot;)] &lt;- &quot;B_cell_3&quot;
seuratM$colclusterName[which(seuratM$RNA_snn_res.0.1 == &quot;5&quot;)] &lt;- &quot;NK&quot;
seuratM$colclusterName[which(seuratM$RNA_snn_res.0.1 == &quot;6&quot;)] &lt;- &quot;pDC_1&quot;
seuratM$colclusterName[which(seuratM$RNA_snn_res.0.1 == &quot;7&quot;)] &lt;- &quot;Mph1&quot;
seuratM$colclusterName[which(seuratM$RNA_snn_res.0.1 == &quot;8&quot;)] &lt;- &quot;pDC_2&quot;
seuratM$colclusterName[which(seuratM$RNA_snn_res.0.1 == &quot;9&quot;)] &lt;- &quot;Mph2&quot;
seuratM$colclusterName[which(seuratM$RNA_snn_res.0.1 == &quot;10&quot;)] &lt;- &quot;proliferating&quot;




# Define your desired order
celltype_order &lt;- c(&quot;T_cell_1&quot;,&quot;T_cell_2&quot;,&quot;B_cell_1&quot;,&quot;B_cell_2&quot;,&quot;B_cell_3&quot;,&quot;NK&quot;, &quot;Mph1&quot;,&quot;Mph2&quot;,&quot;proliferating&quot;,&quot;pDC_1&quot;,&quot;pDC_2&quot;)

seuratM$colclusterName &lt;- factor(seuratM$colclusterName, levels = celltype_order)



# Define the order you want
patient_order &lt;- c(&quot;CUP_01&quot;, &quot;CUP_03&quot;, &quot;CUP_04&quot;, &quot;CUP_05&quot;, &quot;hLN_04&quot;,&quot;hLN_05&quot;)  # customize as needed
seuratM$patient &lt;- factor(seuratM$patient, levels = patient_order)

Idents(seuratM) &lt;- seuratM$colclusterName
DimPlot(seuratM, reduction = &quot;umap&quot;, pt.size = 0.1, cols = colclusterName, label =FALSE) </code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/Assign%20cell%20types%20to%20clusters-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seuratM) &lt;- seuratM$colclusterName
DimPlot(seuratM, reduction = &quot;umap&quot;, pt.size = 0.1, cols = colclusterName, label =FALSE, split.by = &quot;patient&quot;) </code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/Assign%20cell%20types%20to%20clusters-2.png" width="672" style="display: block; margin: auto;" /></p>
<div id="gene-expression-in-different-cell-types"
class="section level3 tabset">
<h3 class="tabset">Gene expression in different cell types</h3>
<div id="t-cell-marker" class="section level4">
<h4>T cell marker</h4>
<pre class="r"><code>## T cell marker
FeaturePlot(seuratM, features = &quot;ENSG00000198851.CD3E&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/T%20cell%20marker-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM, features = &quot;ENSG00000167286.CD3D&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/T%20cell%20marker-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM, features = &quot;ENSG00000010610.CD4&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/T%20cell%20marker-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM, features = &quot;ENSG00000153563.CD8A&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/T%20cell%20marker-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="b-cell-marker" class="section level4">
<h4>B cell marker</h4>
<pre class="r"><code>## B cell marker
FeaturePlot(seuratM, features = &quot;ENSG00000105369.CD79A&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/B%20cell%20marker-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM, features = &quot;ENSG00000007312.CD79B&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/B%20cell%20marker-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM, features = &quot;ENSG00000004468.CD38&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/B%20cell%20marker-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM, features = &quot;ENSG00000211898.IGHD&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/B%20cell%20marker-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM, features = &quot;ENSG00000211899.IGHM&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/B%20cell%20marker-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="myeloid-cell-marker" class="section level4">
<h4>Myeloid cell marker</h4>
<pre class="r"><code>## NK cell marker
FeaturePlot(seuratM, features = &quot;ENSG00000134539.KLRD1&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/Myeloid%20cell%20marker-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM, features = &quot;ENSG00000111796.KLRB1&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/Myeloid%20cell%20marker-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM, features = &quot;ENSG00000134545.KLRC1&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/Myeloid%20cell%20marker-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM, features = &quot;ENSG00000211829.TRDC&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/Myeloid%20cell%20marker-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## Macrophage marker
FeaturePlot(seuratM, features = &quot;ENSG00000090382.LYZ&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/Myeloid%20cell%20marker-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM, features = &quot;ENSG00000085265.FCN1&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/Myeloid%20cell%20marker-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM, features = &quot;ENSG00000163220.S100A9&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/Myeloid%20cell%20marker-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM, features = &quot;ENSG00000182578.CSF1R&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/Myeloid%20cell%20marker-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM, features = &quot;ENSG00000170458.CD14&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/Myeloid%20cell%20marker-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## pDC
FeaturePlot(seuratM, features = &quot;ENSG00000185507.IRF7&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/Myeloid%20cell%20marker-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM, features = &quot;ENSG00000239961.LILRA4&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/Myeloid%20cell%20marker-11.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM, features = &quot;ENSG00000186818.LILRB4&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/Myeloid%20cell%20marker-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM, features = &quot;ENSG00000198178.CLEC4C&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/Myeloid%20cell%20marker-13.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="inhibitory-molecule" class="section level4">
<h4>Inhibitory molecule</h4>
<pre class="r"><code>## T cell - inhibitory molecule
FeaturePlot(seuratM, features = &quot;ENSG00000163600.ICOS&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/Inhibitory%20molecule-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM, features = &quot;ENSG00000163599.CTLA4&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/Inhibitory%20molecule-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM, features = &quot;ENSG00000188389.PDCD1&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/Inhibitory%20molecule-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM, features = &quot;ENSG00000181847.TIGIT&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/Inhibitory%20molecule-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeaturePlot(seuratM, features = &quot;ENSG00000089692.LAG3&quot;, pt.size = 0.5, cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;))</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/Inhibitory%20molecule-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="plot-gene-signatures-onto-umap" class="section level2">
<h2>Plot gene signatures onto UMAP</h2>
<pre class="r"><code>##convert seurat object to sce object
sce &lt;- as.SingleCellExperiment(seuratM)

signatures &lt;- list(
  T_cell = c(&quot;CD3E&quot;,&quot;CD3D&quot;,&quot;CD4&quot;,&quot;CD8A&quot;),
  B_cell = c(&quot;CD79A&quot;,&quot;CD79B&quot;,&quot;IGHD&quot;,&quot;IGHM&quot;,&quot;JCHAIN&quot;),
  Myeloid_cell = c(&quot;KLRD1&quot;,&quot;KLRB1&quot;,&quot;KLRC1&quot;,&quot;TRDC&quot;,&quot;LYZ&quot;,&quot;FCN1&quot;,&quot;S100A9&quot;,&quot;CSF1R&quot;,&quot;CD14&quot; ,&quot;IRF7&quot;,&quot;LILRA4&quot;,&quot;LILRB4&quot;),
  Inhibitory_function= c(&quot;PDCD1&quot;,&quot;TIGIT&quot;,&quot;CTLA4&quot;,&quot;LAG3&quot;,&quot;ICOS&quot;)
    )


# Extract gene symbols
genes &lt;- data.frame(geneID = rownames(sce)) %&gt;%
  mutate(gene = gsub(&quot;.*\\.&quot;, &quot;&quot;, geneID))

# Function to compute signature and return UMAP plot
plot_signature_umap &lt;- function(sig_name, sig_genes) {
  signGenes &lt;- genes %&gt;% filter(gene %in% sig_genes)
  sceSub &lt;- sce[rownames(sce) %in% signGenes$geneID, ]
  
  cntMat &lt;- rowSums(t(as.matrix(sceSub@assays@data$logcounts))) / nrow(signGenes)
  sceSub$sign &lt;- cntMat
  sceSub$sign2 &lt;- pmin(sceSub$sign, 2.5)  # cap at 2.5
  
  plotUMAP(sceSub, colour_by = &quot;sign2&quot;, point_size = 1) +
    scale_color_viridis_c() +  # or another color scale
    ggtitle(paste0(sig_name, &quot; signature&quot;)) +
    theme_minimal() +
    theme(legend.position = &quot;right&quot;)
}

# Generate all plots
umap_plots &lt;- lapply(names(signatures), function(name) {
  plot_signature_umap(name, signatures[[name]])
})

# Optionally: show all plots (in interactive session or PDF)
# Print them one by one
for (p in umap_plots) print(p)</code></pre>
<p><img src="figure/CUP_CD45_merge_healthy.Rmd/cell%20type%20gene%20signatures-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/CUP_CD45_merge_healthy.Rmd/cell%20type%20gene%20signatures-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/CUP_CD45_merge_healthy.Rmd/cell%20type%20gene%20signatures-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/CUP_CD45_merge_healthy.Rmd/cell%20type%20gene%20signatures-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>### save seurat object
saveRDS(seuratM, file=&quot;/R/CUP/data/CD45pos_seurat_merge.rds&quot;)</code></pre>
</div>
<div id="session-info" class="section level2">
<h2>session info</h2>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Tue Jul 22 22:54:50 2025&quot;</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 11 x64 (build 22631)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8  LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] RColorBrewer_1.1-3          NCmisc_1.2.0                VennDiagram_1.7.3          
 [4] futile.logger_1.4.3         ggupset_0.4.1               gridExtra_2.3              
 [7] DOSE_3.30.5                 enrichplot_1.24.4           msigdbr_24.1.0             
[10] org.Hs.eg.db_3.19.1         AnnotationDbi_1.66.0        clusterProfiler_4.12.6     
[13] multtest_2.60.0             metap_1.12                  scater_1.32.1              
[16] scuttle_1.14.0              destiny_3.18.0              circlize_0.4.16            
[19] muscat_1.18.0               viridis_0.6.5               viridisLite_0.4.2          
[22] lubridate_1.9.4             forcats_1.0.0               stringr_1.5.1              
[25] purrr_1.0.4                 readr_2.1.5                 tidyr_1.3.1                
[28] tibble_3.2.1                tidyverse_2.0.0             dplyr_1.1.4                
[31] SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0 Biobase_2.64.0             
[34] GenomicRanges_1.56.2        GenomeInfoDb_1.40.1         IRanges_2.38.1             
[37] S4Vectors_0.42.1            BiocGenerics_0.50.0         MatrixGenerics_1.16.0      
[40] matrixStats_1.5.0           pheatmap_1.0.13             ggpubr_0.6.1               
[43] ggplot2_3.5.2               Seurat_5.3.0                SeuratObject_5.1.0         
[46] sp_2.2-0                    runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] igraph_2.1.4              ica_1.0-3                 plotly_4.11.0            
  [4] Formula_1.2-5             zlibbioc_1.50.0           tidyselect_1.2.1         
  [7] bit_4.6.0                 doParallel_1.0.17         clue_0.3-66              
 [10] lattice_0.22-6            rjson_0.2.23              blob_1.2.4               
 [13] S4Arrays_1.4.1            pbkrtest_0.5.4            parallel_4.4.1           
 [16] png_0.1-8                 plotrix_3.8-4             cli_3.6.3                
 [19] ggplotify_0.1.2           goftest_1.2-3             VIM_6.2.2                
 [22] variancePartition_1.34.0  BiocNeighbors_1.22.0      shadowtext_0.1.5         
 [25] uwot_0.2.3                curl_6.4.0                tidytree_0.4.6           
 [28] mime_0.12                 evaluate_1.0.4            ComplexHeatmap_2.20.0    
 [31] stringi_1.8.7             backports_1.5.0           lmerTest_3.1-3           
 [34] qqconf_1.3.2              httpuv_1.6.15             magrittr_2.0.3           
 [37] rappdirs_0.3.3            splines_4.4.1             ggraph_2.2.1             
 [40] sctransform_0.4.2         ggbeeswarm_0.7.2          DBI_1.2.3                
 [43] jquerylib_0.1.4           smoother_1.3              withr_3.0.2              
 [46] git2r_0.36.2              corpcor_1.6.10            reformulas_0.4.1         
 [49] class_7.3-22              rprojroot_2.0.4           lmtest_0.9-40            
 [52] tidygraph_1.3.1           formatR_1.14              colourpicker_1.3.0       
 [55] htmlwidgets_1.6.4         fs_1.6.6                  ggrepel_0.9.6            
 [58] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [61] DESeq2_1.44.0             ranger_0.17.0             DEoptimR_1.1-3-1         
 [64] reticulate_1.42.0         hexbin_1.28.5             zoo_1.8-14               
 [67] XVector_0.44.0            knitr_1.50                ggplot.multistats_1.0.1  
 [70] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [73] foreach_1.5.2             patchwork_1.3.1           caTools_1.18.3           
 [76] data.table_1.17.6         ggtree_3.12.0             R.oo_1.27.1              
 [79] RSpectra_0.16-2           irlba_2.3.5.1             fastDummies_1.7.5        
 [82] gridGraphics_0.5-1        lazyeval_0.2.2            yaml_2.3.10              
 [85] survival_3.6-4            scattermore_1.2           crayon_1.5.3             
 [88] RcppAnnoy_0.0.22          progressr_0.15.1          tweenr_2.0.3             
 [91] later_1.3.2               ggridges_0.5.6            codetools_0.2-20         
 [94] GlobalOptions_0.1.2       aod_1.3.3                 KEGGREST_1.44.1          
 [97] Rtsne_0.17                shape_1.4.6.1             limma_3.60.6             
[100] pkgconfig_2.0.3           TMB_1.9.17                spatstat.univar_3.1-3    
[103] mathjaxr_1.8-0            EnvStats_3.1.0            aplot_0.2.7              
[106] scatterplot3d_0.3-44      ape_5.8-1                 spatstat.sparse_3.1-0    
[109] xtable_1.8-4              car_3.1-3                 plyr_1.8.9               
[112] httr_1.4.7                rbibutils_2.3             tools_4.4.1              
[115] globals_0.18.0            beeswarm_0.4.0            broom_1.0.8              
[118] nlme_3.1-164              lambda.r_1.2.4            assertthat_0.2.1         
[121] lme4_1.1-37               digest_0.6.36             numDeriv_2016.8-1.1      
[124] Matrix_1.7-0              farver_2.1.2              tzdb_0.4.0               
[127] remaCor_0.0.18            reshape2_1.4.4            yulab.utils_0.2.0        
[130] glue_1.7.0                cachem_1.1.0              polyclip_1.10-7          
[133] generics_0.1.4            Biostrings_2.72.1         mvtnorm_1.3-3            
[136] parallelly_1.45.0         mnormt_2.1.1              statmod_1.5.0            
[139] RcppHNSW_0.6.0            ScaledMatrix_1.12.0       carData_3.0-5            
[142] minqa_1.2.8               pbapply_1.7-2             httr2_1.1.2              
[145] spam_2.11-1               gson_0.1.0                graphlayouts_1.2.2       
[148] gtools_3.9.5              ggsignif_0.6.4            RcppEigen_0.3.4.0.2      
[151] shiny_1.11.0              GenomeInfoDbData_1.2.12   glmmTMB_1.1.11           
[154] R.utils_2.13.0            memoise_2.0.1             rmarkdown_2.29           
[157] scales_1.4.0              R.methodsS3_1.8.2         future_1.58.0            
[160] RANN_2.6.1                Cairo_1.6-2               spatstat.data_3.1-6      
[163] rstudioapi_0.17.1         cluster_2.1.6             mutoss_0.1-13            
[166] spatstat.utils_3.1-4      hms_1.1.3                 fitdistrplus_1.2-3       
[169] cowplot_1.1.3             colorspace_2.1-1          rlang_1.1.4              
[172] DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0  xts_0.14.1               
[175] dotCall64_1.2             shinydashboard_0.7.3      ggforce_0.5.0            
[178] laeken_0.5.3              mgcv_1.9-1                xfun_0.52                
[181] e1071_1.7-16              TH.data_1.1-3             iterators_1.0.14         
[184] abind_1.4-8               GOSemSim_2.30.2           treeio_1.28.0            
[187] futile.options_1.0.1      bitops_1.0-9              Rdpack_2.6.4             
[190] promises_1.3.3            scatterpie_0.2.5          RSQLite_2.4.1            
[193] qvalue_2.36.0             sandwich_3.1-1            fgsea_1.30.0             
[196] DelayedArray_0.30.1       proxy_0.4-27              GO.db_3.19.1             
[199] compiler_4.4.1            prettyunits_1.2.0         boot_1.3-30              
[202] beachmat_2.20.0           listenv_0.9.1             Rcpp_1.0.13              
[205] edgeR_4.2.2               workflowr_1.7.1           BiocSingular_1.20.0      
[208] tensor_1.5.1              MASS_7.3-60.2             progress_1.2.3           
[211] BiocParallel_1.38.0       babelgene_22.9            spatstat.random_3.4-1    
[214] R6_2.6.1                  fastmap_1.2.0             multcomp_1.4-28          
[217] fastmatch_1.1-6           rstatix_0.7.2             vipor_0.4.7              
[220] TTR_0.24.4                ROCR_1.0-11               TFisher_0.2.0            
[223] rsvd_1.0.5                vcd_1.4-13                nnet_7.3-19              
[226] gtable_0.3.6              KernSmooth_2.23-24        miniUI_0.1.2             
[229] deldir_2.0-4              htmltools_0.5.8.1         ggthemes_5.1.0           
[232] bit64_4.6.0-1             spatstat.explore_3.4-3    lifecycle_1.0.4          
[235] blme_1.0-6                S7_0.2.0                  nloptr_2.2.1             
[238] sass_0.4.10               vctrs_0.6.5               robustbase_0.99-4-1      
[241] spatstat.geom_3.4-1       sn_2.1.1                  ggfun_0.1.9              
[244] future.apply_1.20.0       bslib_0.9.0               pillar_1.10.2            
[247] gplots_3.2.0              pcaMethods_1.96.0         locfit_1.5-9.12          
[250] jsonlite_2.0.0            GetoptLong_1.0.5         </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 11 x64 (build 22631)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8  LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] RColorBrewer_1.1-3          NCmisc_1.2.0                VennDiagram_1.7.3          
 [4] futile.logger_1.4.3         ggupset_0.4.1               gridExtra_2.3              
 [7] DOSE_3.30.5                 enrichplot_1.24.4           msigdbr_24.1.0             
[10] org.Hs.eg.db_3.19.1         AnnotationDbi_1.66.0        clusterProfiler_4.12.6     
[13] multtest_2.60.0             metap_1.12                  scater_1.32.1              
[16] scuttle_1.14.0              destiny_3.18.0              circlize_0.4.16            
[19] muscat_1.18.0               viridis_0.6.5               viridisLite_0.4.2          
[22] lubridate_1.9.4             forcats_1.0.0               stringr_1.5.1              
[25] purrr_1.0.4                 readr_2.1.5                 tidyr_1.3.1                
[28] tibble_3.2.1                tidyverse_2.0.0             dplyr_1.1.4                
[31] SingleCellExperiment_1.26.0 SummarizedExperiment_1.34.0 Biobase_2.64.0             
[34] GenomicRanges_1.56.2        GenomeInfoDb_1.40.1         IRanges_2.38.1             
[37] S4Vectors_0.42.1            BiocGenerics_0.50.0         MatrixGenerics_1.16.0      
[40] matrixStats_1.5.0           pheatmap_1.0.13             ggpubr_0.6.1               
[43] ggplot2_3.5.2               Seurat_5.3.0                SeuratObject_5.1.0         
[46] sp_2.2-0                    runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] igraph_2.1.4              ica_1.0-3                 plotly_4.11.0            
  [4] Formula_1.2-5             zlibbioc_1.50.0           tidyselect_1.2.1         
  [7] bit_4.6.0                 doParallel_1.0.17         clue_0.3-66              
 [10] lattice_0.22-6            rjson_0.2.23              blob_1.2.4               
 [13] S4Arrays_1.4.1            pbkrtest_0.5.4            parallel_4.4.1           
 [16] png_0.1-8                 plotrix_3.8-4             cli_3.6.3                
 [19] ggplotify_0.1.2           goftest_1.2-3             VIM_6.2.2                
 [22] variancePartition_1.34.0  BiocNeighbors_1.22.0      shadowtext_0.1.5         
 [25] uwot_0.2.3                curl_6.4.0                tidytree_0.4.6           
 [28] mime_0.12                 evaluate_1.0.4            ComplexHeatmap_2.20.0    
 [31] stringi_1.8.7             backports_1.5.0           lmerTest_3.1-3           
 [34] qqconf_1.3.2              httpuv_1.6.15             magrittr_2.0.3           
 [37] rappdirs_0.3.3            splines_4.4.1             ggraph_2.2.1             
 [40] sctransform_0.4.2         ggbeeswarm_0.7.2          DBI_1.2.3                
 [43] jquerylib_0.1.4           smoother_1.3              withr_3.0.2              
 [46] git2r_0.36.2              corpcor_1.6.10            reformulas_0.4.1         
 [49] class_7.3-22              rprojroot_2.0.4           lmtest_0.9-40            
 [52] tidygraph_1.3.1           formatR_1.14              colourpicker_1.3.0       
 [55] htmlwidgets_1.6.4         fs_1.6.6                  ggrepel_0.9.6            
 [58] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [61] DESeq2_1.44.0             ranger_0.17.0             DEoptimR_1.1-3-1         
 [64] reticulate_1.42.0         hexbin_1.28.5             zoo_1.8-14               
 [67] XVector_0.44.0            knitr_1.50                ggplot.multistats_1.0.1  
 [70] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [73] foreach_1.5.2             patchwork_1.3.1           caTools_1.18.3           
 [76] data.table_1.17.6         ggtree_3.12.0             R.oo_1.27.1              
 [79] RSpectra_0.16-2           irlba_2.3.5.1             fastDummies_1.7.5        
 [82] gridGraphics_0.5-1        lazyeval_0.2.2            yaml_2.3.10              
 [85] survival_3.6-4            scattermore_1.2           crayon_1.5.3             
 [88] RcppAnnoy_0.0.22          progressr_0.15.1          tweenr_2.0.3             
 [91] later_1.3.2               ggridges_0.5.6            codetools_0.2-20         
 [94] GlobalOptions_0.1.2       aod_1.3.3                 KEGGREST_1.44.1          
 [97] Rtsne_0.17                shape_1.4.6.1             limma_3.60.6             
[100] pkgconfig_2.0.3           TMB_1.9.17                spatstat.univar_3.1-3    
[103] mathjaxr_1.8-0            EnvStats_3.1.0            aplot_0.2.7              
[106] scatterplot3d_0.3-44      ape_5.8-1                 spatstat.sparse_3.1-0    
[109] xtable_1.8-4              car_3.1-3                 plyr_1.8.9               
[112] httr_1.4.7                rbibutils_2.3             tools_4.4.1              
[115] globals_0.18.0            beeswarm_0.4.0            broom_1.0.8              
[118] nlme_3.1-164              lambda.r_1.2.4            assertthat_0.2.1         
[121] lme4_1.1-37               digest_0.6.36             numDeriv_2016.8-1.1      
[124] Matrix_1.7-0              farver_2.1.2              tzdb_0.4.0               
[127] remaCor_0.0.18            reshape2_1.4.4            yulab.utils_0.2.0        
[130] glue_1.7.0                cachem_1.1.0              polyclip_1.10-7          
[133] generics_0.1.4            Biostrings_2.72.1         mvtnorm_1.3-3            
[136] parallelly_1.45.0         mnormt_2.1.1              statmod_1.5.0            
[139] RcppHNSW_0.6.0            ScaledMatrix_1.12.0       carData_3.0-5            
[142] minqa_1.2.8               pbapply_1.7-2             httr2_1.1.2              
[145] spam_2.11-1               gson_0.1.0                graphlayouts_1.2.2       
[148] gtools_3.9.5              ggsignif_0.6.4            RcppEigen_0.3.4.0.2      
[151] shiny_1.11.0              GenomeInfoDbData_1.2.12   glmmTMB_1.1.11           
[154] R.utils_2.13.0            memoise_2.0.1             rmarkdown_2.29           
[157] scales_1.4.0              R.methodsS3_1.8.2         future_1.58.0            
[160] RANN_2.6.1                Cairo_1.6-2               spatstat.data_3.1-6      
[163] rstudioapi_0.17.1         cluster_2.1.6             mutoss_0.1-13            
[166] spatstat.utils_3.1-4      hms_1.1.3                 fitdistrplus_1.2-3       
[169] cowplot_1.1.3             colorspace_2.1-1          rlang_1.1.4              
[172] DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0  xts_0.14.1               
[175] dotCall64_1.2             shinydashboard_0.7.3      ggforce_0.5.0            
[178] laeken_0.5.3              mgcv_1.9-1                xfun_0.52                
[181] e1071_1.7-16              TH.data_1.1-3             iterators_1.0.14         
[184] abind_1.4-8               GOSemSim_2.30.2           treeio_1.28.0            
[187] futile.options_1.0.1      bitops_1.0-9              Rdpack_2.6.4             
[190] promises_1.3.3            scatterpie_0.2.5          RSQLite_2.4.1            
[193] qvalue_2.36.0             sandwich_3.1-1            fgsea_1.30.0             
[196] DelayedArray_0.30.1       proxy_0.4-27              GO.db_3.19.1             
[199] compiler_4.4.1            prettyunits_1.2.0         boot_1.3-30              
[202] beachmat_2.20.0           listenv_0.9.1             Rcpp_1.0.13              
[205] edgeR_4.2.2               workflowr_1.7.1           BiocSingular_1.20.0      
[208] tensor_1.5.1              MASS_7.3-60.2             progress_1.2.3           
[211] BiocParallel_1.38.0       babelgene_22.9            spatstat.random_3.4-1    
[214] R6_2.6.1                  fastmap_1.2.0             multcomp_1.4-28          
[217] fastmatch_1.1-6           rstatix_0.7.2             vipor_0.4.7              
[220] TTR_0.24.4                ROCR_1.0-11               TFisher_0.2.0            
[223] rsvd_1.0.5                vcd_1.4-13                nnet_7.3-19              
[226] gtable_0.3.6              KernSmooth_2.23-24        miniUI_0.1.2             
[229] deldir_2.0-4              htmltools_0.5.8.1         ggthemes_5.1.0           
[232] bit64_4.6.0-1             spatstat.explore_3.4-3    lifecycle_1.0.4          
[235] blme_1.0-6                S7_0.2.0                  nloptr_2.2.1             
[238] sass_0.4.10               vctrs_0.6.5               robustbase_0.99-4-1      
[241] spatstat.geom_3.4-1       sn_2.1.1                  ggfun_0.1.9              
[244] future.apply_1.20.0       bslib_0.9.0               pillar_1.10.2            
[247] gplots_3.2.0              pcaMethods_1.96.0         locfit_1.5-9.12          
[250] jsonlite_2.0.0            GetoptLong_1.0.5         </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
